src/components/AttachmentPicker/index.native.js

```diff
@@ -129,8 +140,12 @@ class AttachmentPicker extends Component {
             return;
         }
 
-        const result = getDataForUpload(attachment);
-        this.completeAttachmentSelection(result);
+        getDataForUpload(attachment).then((result) => {
```

Comment:
What is the use of throwing this error in a catch block?
```suggestion
        reuturn getDataForUpload(attachment).then((result) => {
```

Reply:
Something I added based on the suggestion here, https://github.com/Expensify/App/pull/6588#discussion_r763960171

Reply:
I meant what is the use of throwing error in promise if you don't pass it down.

Reply:
Well, I didn't return the Promise. I can do it and we won't need the catch block.

Reply:
Updated